### PR TITLE
Added equals operator that can only accept a single value

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description

Most of the rest of the code coming in follow up PR where we add the equals operator to the CreateRule API.

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/authz-service/storage/v2/condition.go [hunk 1]
```diff
@@ -47,12 +47,19 @@ func NewCondition(ruleType RuleType, value []string,
...
+func validateConditionInputs(value []string,
+	attribute ConditionAttribute, ruleType RuleType, operator ConditionOperator) error {
 
 	if len(value) == 0 {
 		return errors.New("a condition must have a value list that is not empty")
 	}
 
+	if operator == Equals {
+		if len(value) != 1 {
+			return errors.New("if the operator is equals, you must have a single entry in the value list")
```
Comment (msorens) [inline components/authz-service/storage/v2/condition.go]: Suggest just: "must specify exactly one value for 'equals' operator"

File: components/authz-service/storage/v2/condition.go [hunk 2]
```diff
@@ -47,12 +47,19 @@ func NewCondition(ruleType RuleType, value []string,
 	}, nil
 }
 
-func validateConditionInputs(value []string, attribute ConditionAttribute, ruleType RuleType) error {
+func validateConditionInputs(value []string,
+	attribute ConditionAttribute, ruleType RuleType, operator ConditionOperator) error {
 
 	if len(value) == 0 {
 		return errors.New("a condition must have a value list that is not empty")
```
Comment (msorens) [inline components/authz-service/storage/v2/condition.go]: Suggest: "must specify one or more values"

File: components/authz-service/storage/v2/postgres/postgres_test.go [hunk 3]
```diff
@@ -3333,6 +3333,17 @@ func TestCreateRule(t *testing.T) {
...
+			condition1, err := storage.NewCondition(storage.Event, []string{}, storage.ChefServer, storage.Equals)
+			assert.Equal(t, storage.Condition{}, condition1)
+			assert.Error(t, err)
+		},
+		"creating an equals condition with multiple entries returns an error": func(t *testing.T) {
+			condition1, err := storage.NewCondition(storage.Event,
+				[]string{"chef-server-1", "chef-server-2"}, storage.ChefServer, storage.Equals)
+			assert.Equal(t, storage.Condition{}, condition1)
+			assert.Error(t, err)
+		},
```
Comment (msorens) [inline components/authz-service/storage/v2/postgres/postgres_test.go]: Suggest a couple more tests:
"multiple entries for a 'member-of' condition are allowed"
"single entry for a 'member-of' condition is allowed"

File: components/authz-service/storage/v2/postgres/postgres_test.go [hunk 4]
```diff
@@ -3333,6 +3333,17 @@ func TestCreateRule(t *testing.T) {
 			_, err := storage.NewRule("new-id-1", projID, "name", storage.Node, []storage.Condition{})
 			assert.Error(t, err)
 		},
+		"creating a condition with zero entries returns an error": func(t *testing.T) {
```
Comment (msorens) [inline components/authz-service/storage/v2/postgres/postgres_test.go]: Suggest explicitly say "for 'equals' operator" in this test,
and add another test for the member-of operator.

File: components/authz-service/storage/v2/postgres/postgres_test.go (hunk 3 above)
Comment (tylercloke) [reply components/authz-service/storage/v2/postgres/postgres_test.go]: We have a million tests that cover this case but I can add them explicitly.

**Pull request merged**